core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/ArithmeticFunction.java

```diff
@@ -43,168 +44,255 @@ public class ArithmeticFunction {
-    repository.register(subtract());
-    repository.register(multiply());
+    repository.register(addFunction());
     repository.register(divide());
-    repository.register(modules());
+    repository.register(divideFunction());
+    repository.register(mod());
+    repository.register(modulus());
+    repository.register(modulusFunction());
+    repository.register(multiply());
+    repository.register(multiplyFunction());
+    repository.register(subtract());
+    repository.register(subtractFunction());
   }
 
-  private static DefaultFunctionResolver add() {
-    return FunctionDSL.define(BuiltinFunctionName.ADD.getName(),
-        FunctionDSL.impl(
-            FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprByteValue(v1.byteValue() + v2.byteValue())),
-            BYTE, BYTE, BYTE),
-        FunctionDSL.impl(
-            FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprShortValue(v1.shortValue() + v2.shortValue())),
-            SHORT, SHORT, SHORT),
-        FunctionDSL.impl(
-            FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprIntegerValue(Math.addExact(v1.integerValue(),
-                    v2.integerValue()))),
-            INTEGER, INTEGER, INTEGER),
-        FunctionDSL.impl(
-            FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprLongValue(Math.addExact(v1.longValue(), v2.longValue()))),
-            LONG, LONG, LONG),
-        FunctionDSL.impl(
-            FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprFloatValue(v1.floatValue() + v2.floatValue())),
-            FLOAT, FLOAT, FLOAT),
-        FunctionDSL.impl(
-            FunctionDSL.nullMissingHandling(
-                (v1, v2) -> new ExprDoubleValue(v1.doubleValue() + v2.doubleValue())),
-            DOUBLE, DOUBLE, DOUBLE)
+  /**
+   * Definition of add(x, y) function.
+   * Returns the number x plus number y
+   * The supported signature of add function is
+   * (x: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE, y: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE)
+   * -> wider type between types of x and y
+   */
+  private static DefaultFunctionResolver addBase(FunctionName functionName) {
+    return FunctionDSL.define(functionName,
+            FunctionDSL.impl(
+                    FunctionDSL.nullMissingHandling(
+                            (v1, v2) -> new ExprByteValue(v1.byteValue() + v2.byteValue())),
+                    BYTE, BYTE, BYTE),
+            FunctionDSL.impl(
+                    FunctionDSL.nullMissingHandling(
+                            (v1, v2) -> new ExprShortValue(v1.shortValue() + v2.shortValue())),
+                    SHORT, SHORT, SHORT),
+            FunctionDSL.impl(
+                    FunctionDSL.nullMissingHandling(
+                            (v1, v2) -> new ExprIntegerValue(Math.addExact(v1.integerValue(),
+                                    v2.integerValue()))),
+                    INTEGER, INTEGER, INTEGER),
```

Comment:
Please, recover the fix for integer overflow (and other types too) - for `add`, `subtract` and `multiply` both

Reply:
I believe that it should be left as is for a few reasons. As it is currently it is replicating the functionality of the functions as they are in the previous engine. Also this is how these functions work in Java, and almost every other language. For example could you imagine adding two integers together and the output being a long, it overcomplicates things and as always with overflow errors the burden should be on the user to use the right data type for the size of data they are working with

Reply:
Ok
Just open a GH bug for overflow issue to ensure that it is not forgotten

Reply:
Is it really an issue? To me overflowing on math functions are expected behaviour. If this is a bug then what would the hypothetical solution be, because changing the data type that the user gets back from the one they were expecting doesn't seem right to me.

Reply:
It could be closed as 'won't fix' if not